lib/message_bus/codec/oj_fast.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+require 'oj' unless defined? ::Oj
+require 'base64'
+
+module MessageBus
+  module Codec
+    class FastIdList
+      def self.from_array(array)
+        new(array.sort.pack("V*"))
+      end
+
+      def self.from_string(string)
+        new(string)
+      end
+
+      def initialize(packed)
+        @packed = packed
+      end
+
+      def include?(id)
+        found = (0...length).bsearch do |index|
+          @packed.byteslice(index * 4, 4).unpack1("V") >= id
+        end
+
+        if found
+          found && @packed.byteslice(found * 4, 4).unpack1("V") == id
+        end
```

Comment:
TIL how to pack, thanks for educating me :)

Question though, you mentioned the problem with 4 bytes, but why is it not possible to use `"Q"` to pack 8 bytes?

I made a test and it seems to work:

```rb
numbers = [1,999999999999999].pack("Q*")
(0...numbers.length).step(8).map { |index| numbers.byteslice(index, 8).unpack1("Q") }
=> [1, 999999999999999]
```

Sorry if that doesn't make sense :)

Reply:
Sure made Q vs V optional in the demo codec in the bench directory.